### PR TITLE
Sprint 35

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 35

Known issues currently being addressed:

* Security vulnerability in `serverless` https://github.com/serverless/serverless/issues/7402

Before deploying to AWS, please reset DynamoDB and Elasticsearch with `clear-env.sh`

### CODE REVIEW

**Comment by kfoley-18F** [comment]
We use this list when performing a code review to ensure that all tasks have been completed.

- [ ] begin by deleting any existing copy of the Git repository, running `docker system prune` to remove any cached artifacts from any prior code review
- [ ] clone a fresh copy of the Git repository (`git clone https://github.com/ustaxcourt/ef-cms.git`), fetch the pull request and name it for the sprint (e.g., `git fetch origin pull/204/head:sprint-43`), and then switch to that branch (e.g. `git checkout sprint-43`)
- [ ] review the pull request itself, to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] stand up the site locally, with `./docker-run.sh`
	- [ ] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [ ] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [ ] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
	- [ ] methodically review all new tests for correctness, quality of naming
- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Comment by kfoley-18F** [comment]
The SonarCloud results seem to be failing or broken. Is anybody looking into that?

**Comment by matthopson** [comment]
> The SonarCloud results seem to be failing or broken. Is anybody looking into that?

Yes.

**Comment by kfoley-18F** [comment]
Thanks @matthopson for the update for SonarCloud!

File: shared/src/business/useCases/createCaseFromPaperInteractor.js [hunk 1]
```diff
@@ -293,17 +295,17 @@ exports.createCaseFromPaperInteractor = async ({
...
-      applicationContext,
-      caseToCreate: caseToAdd.validate().toRawObject(),
-    }),
-    applicationContext.getPersistenceGateway().saveWorkItemForPaper({
-      applicationContext,
-      messageId: newMessage.messageId,
-      workItem: newWorkItem.validate().toRawObject(),
-    }),
-  ]);
+  // Warning - do not use Promise.all() as it seems to cause intermittent failures
```
[the remark this answers is not in the document]
Comment (matthopson) [reply shared/src/business/useCases/createCaseFromPaperInteractor.js]: We're uncovering some unexpected behavior at times when using `Promise.all()`- we had a mob session yesterday, where we learned a little more about it, but can't say for sure if it is the thing that is causing the issues, but semi-related. The warning was added to curtail its use until we can know for sure. Essentially, you can create some race conditions if you're not careful in certain situations. We'd like to document this better and remove the warning when we're 100% sure what's going on.

**Comment by waldoj** [comment]
@JessicaMarine, @kfoley-18F has reviewed this work and recommends that it be merged into `staging`.

**Pull request merged**